lib/utils/descriptor.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * descriptor.js - Output descriptor for hsd
+ * Copyright (c) 2020, The Handshake Developers (MIT License).
+ * Copyright (c) 2020, Mark Tyneway (MIT License).
+ * https://github.com/handshake-org/hsd
+ */
+
+'use strict';
+
+const assert = require('bsert');
+const Script = require('../script/script');
+const secp256k1 = require('bcrypto/lib/secp256k1');
+const Address = require('../primitives/address');
+const {BufferSet} = require('buffer-map');
+
+// all are top level
+const typeCache = new Set([
+  'wpkh',
+  'wsh',
+  'addr',
+  'opreturn'
+]);
+
+class Descriptor {
+  constructor(descriptor) {
+    this.descriptor = descriptor || '';
```

Comment:
Does this syntax just cast to string?

Reply:
If `descriptor` is falsy, `null` or `undefined`, then assign the `''` to `this.descriptor`. It is expected that `descriptor` is a string.

Reply:
ahhhhhh gotcha. would it be more hsd style like

```
this.descriptor = '';

if (descriptor != null) {
  assert(typeof descriptor === 'String');
  this.descriptor = descriptor;
}
```
etc?

Reply:
I skipped implementing the `fromOptions` pattern because it felt kind of verbose.

You can see other places that lack it in the codebase, although this could be for performance reasons since adding extra methods will increase the size of the object in memory.

https://github.com/handshake-org/hsd/blob/d1b42a6a50332ab64de4750297b468de76b05b76/lib/net/packets.js#L772-L785

This is not a performance critical piece of code, so I could follow the pattern if people have a strong opinion on it.